### PR TITLE
Add `filepath` fields in Python and Rust

### PR DESCRIPTION
**Pull request by sarahyurick**
As part of the dynamic partition pruning optimizer rule, we'll be reading in Parquet files on the Rust side. To do this, we need to obtain information about the file paths, when available, to pass into Rust. Here, I use similar logic as the statistics logic from https://github.com/dask-contrib/dask-sql/pull/1061 and https://github.com/dask-contrib/dask-sql/pull/1027.

### CODE REVIEW

File: dask_sql/context.py [hunk 1]
```diff
@@ -250,6 +250,13 @@ def create_table(
             **kwargs,
         )
 
+        if type(input_table) == str:
+            filepath = input_table
+            dc.filepath = input_table
+            self.schema[schema_name].filepaths[table_name.lower()] = input_table
+        else:
+            filepath = ""
```
Comment (sarahyurick) [inline dask_sql/context.py:258]: Using empty string here as `None` throws an error.

File: dask_sql/context.py [hunk 2]
```diff
@@ -250,6 +250,13 @@ def create_table(
             **kwargs,
         )
 
+        if type(input_table) == str:
+            filepath = input_table
+            dc.filepath = input_table
```
Comment (jdye64) [inline dask_sql/context.py:255]: I like this. I feel like this should have always been an attribute in the DataContainer all along

File: dask_sql/context.py (hunk 1 above)
Comment (jdye64) [reply dask_sql/context.py:258]: I'm fine with `""` but wonder why `None` was causing issues? hmm

Comment (jdye64) [reply dask_sql/context.py:258]: What the error on the Rust side or Python side?

**Review by jdye64 — APPROVED** [review]
Changes look good. Little concerned about using `""` instead of `None` for the "empty" filepath but only from the Rust side as I worry someone might be manually checking for `Result::None` in Rust but `Result::Some("")` would be there but the intention would be meant for the filepath to in fact by "not present" Thanks @sarahyurick

File: dask_sql/context.py (hunk 1 above)
Comment (jdye64) [reply dask_sql/context.py:258]: Ahh, yep. That is worth doing I think. Thanks for pointing that out Charles.

Comment (sarahyurick) [reply dask_sql/context.py]: Thanks @charlesbluca for looking into this!

**Comment by sarahyurick** [comment]
Thanks @jdye64 !

**Review by charlesbluca — APPROVED** [review]
Thanks @sarahyurick!

File: dask_planner/src/sql.rs [hunk 3]
```diff
@@ -141,12 +141,25 @@ impl ContextProvider for DaskSQLContext {
...
                             .unwrap()
                             .statistics;
+                        let filepath = &self
+                            .schemas
+                            .get(reference.schema.as_ref())
+                            .unwrap()
+                            .tables
+                            .get(reference.table.as_ref())
+                            .unwrap()
+                            .filepath;
```
Comment (ayushdg) [inline dask_planner/src/sql.rs:151]: Minor nit: Can re-use the daskTable reference we get after `unwrap` and re-use it for both statistics and filepath.

File: tests/unit/test_context.py [hunk 4]
```diff
@@ -329,3 +330,29 @@ def test_alter_table(c, df_simple):
...
+            ]
+            * 5,
+            "index": range(15),
+        },
+    )
+    dd.from_pandas(df, npartitions=3).to_parquet(parquet_path)
+
+    c.create_table("df", parquet_path, format="parquet")
+
+    assert c.schema["root"].tables["df"].filepath == parquet_path
```
Comment (ayushdg) [inline tests/unit/test_context.py:357]: Minor suggestion: Could we also add a test for a case where the filepath isn't available and the values here would be `None`

**Review by ayushdg — CHANGES_REQUESTED** [review]
Minor suggestions. Overall lgtm, thanks!

**Comment by sarahyurick** [comment]
Thanks @ayushdg @charlesbluca ! Added the suggested changes, lmk what you think.

**Review by ayushdg — APPROVED** [review]
